Assets/_BForBoss/_Weapons/Scripts/EquipmentBehaviour.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                _reloadInputAction.Enable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Enable();
+            }
+
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Enable();
+            }
+
+            _isMouseScrollEnabled = true;
+        }
+        
+        private void DisableEquipmentPlayerInput()
+        {
+            if (_reloadInputAction != null)
+            {
+                _reloadInputAction.Disable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Disable();
+            }
+            
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Disable();
+            }
+
+            _isMouseScrollEnabled = false;
+        }
+
+        private void SetupWeapons()
+        {
+            foreach (var weapon in _weapons)
+            {
+                weapon.Initialize(_fireInputAction, _reloadInputAction);
+                weapon.enabled = false;
+            }
+
+            _weapons[_currentWeaponIndex].enabled = true;
+        }
+        
+        private void SetupPlayerEquipmentInput()
+        {
+            _reloadInputAction = _inputActions.FindAction("Reload");
+            _fireInputAction = _inputActions.FindAction("Fire");
+            _swapWeaponInputAction = _inputActions.FindAction("WeaponSwap");
+        }
+
+        private void SetupControllerSwapWeaponInputBinding()
+        {
+            _swapWeaponInputAction.started += OnControllerSwapWeaponAction;
+        }
+        
+        private void ScrollSwapWeapons(bool isUpwards)
+        {
+            _weapons[_currentWeaponIndex].enabled = false;
+            UpdateCurrentWeaponIndex(isUpwards);
+            _weapons[_currentWeaponIndex].enabled = true;
+        }
+        
+        private void UpdateCurrentWeaponIndex(bool isUpwards)
+        {
+            var indexLength = _weapons.Length - 1;
+            _currentWeaponIndex += isUpwards ? 1 : -1;
+            
+            if (_currentWeaponIndex > indexLength)
+            {
+                _currentWeaponIndex = 0;
+            } 
+            else if (_currentWeaponIndex < 0)
+            {
+                _currentWeaponIndex = indexLength;
+            }
+        }
+
+
+
+        private void Update()
+        {
+            OnMouseSwapWeaponAction();
+        }
+
+        private void Start()
+        {
+            Initialize();
+        }
+
+        private void Awake()
+        {
+            if (_inputActions == null)
+            {
+                Debug.LogWarning("Input Action Asset is missing from equipment behaviour");
+            }
+
+            if (_weapons.IsNullOrEmpty())
+            {
+                Debug.LogWarning("There are currently no weapons equipped within Equipment Behaviour");
+            }
```

Comment:
More of question, how do we feel about using the [OnValidate()](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnValidate.html) Unity Message for null checking components?  Just an idea

So that checking every awake, we ensure the devs make sure no component is null

Reply:
yeah we could, either would work. Especially for value changes on inspector
Might be cleaner overall to separate from `Awake()`
https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnValidate.html